python_modules/dagster/dagster/_core/utils.py

```diff
@@ -127,6 +128,35 @@ def parse_env_var(env_var_str: str) -> Tuple[str, str]:
 class InheritContextThreadPoolExecutor(ThreadPoolExecutor):
     """A ThreadPoolExecutor that copies over contextvars at submit time."""
 
+    def __init__(
+        self,
+        max_workers: int | None = None,
+        thread_name_prefix: str = "",
+        initializer: Callable[..., object] | None = None,
+        initargs: tuple[Any, ...] = ...,
+    ) -> None:
+        super().__init__(max_workers, thread_name_prefix, initializer, initargs)
+        self._futures = []
```

Comment:
Let's add a comment here about why we're doing this. Also, maybe use `_all_futures`? `_in_progress_futures` (encapsulates queued and running status?)